integration_tests/tests/integration/common.rs

```diff
@@ -141,32 +143,26 @@ where
     // An extra retry loop is needed here because our collect request may race against the
     // aggregation job creator, which is responsible for assigning reports to batches in fixed-
     // size tasks.
-    let backoff = ExponentialBackoffBuilder::new()
-        .with_initial_interval(time::Duration::from_millis(500))
-        .with_max_interval(time::Duration::from_millis(500))
-        .with_max_elapsed_time(Some(time::Duration::from_secs(5)))
+    let backoff = ConstantBuilder::new()
+        .with_delay(time::Duration::from_millis(500))
+        .with_max_times(10)
         .build();
-    retry(backoff, || {
+    (|| {
         let query = query.clone();
         async move {
-            match collector.collect(query, aggregation_parameter).await {
-                Ok(collection) => Ok(collection),
-                Err(janus_collector::Error::Http(error_response)) => {
-                    if let Some(DapProblemType::InvalidBatchSize) =
-                        error_response.dap_problem_type()
-                    {
-                        Err(backoff::Error::transient(janus_collector::Error::Http(
-                            error_response,
-                        )))
-                    } else {
-                        Err(backoff::Error::permanent(janus_collector::Error::Http(
-                            error_response,
-                        )))
-                    }
-                }
-                Err(error) => Err(backoff::Error::permanent(error)),
+            collector
+                .collect(query.clone(), aggregation_parameter)
+                .await
+        }
+    })
+    .retry(backoff)
+    .when(|e| {
+        if let janus_collector::Error::Http(error_response) = e {
+            if let Some(DapProblemType::InvalidBatchSize) = error_response.dap_problem_type() {
+                return true;
             }
         }
+        false
```

Comment:
This block screams to me that there must be some clever way to do this without indentation, but I haven't figured it out yet. Some kind of `match`? Clever use of `matches!`?

Reply:
Maybe
```rust
if let janus_collector::Error::Http(error_response) = e {
            return error_response.dap_problem_type()
                     .is_some_and(|t| t==DapProblemType::InvalidBatchSize);
}
```

Reply:
Rather than use a combinator, I'd recommend changing the outer if statement to let-else, and doing an early return in the else block.

Reply:
Code golf: I believe this [match guard](https://doc.rust-lang.org/book/ch19-03-pattern-syntax.html#extra-conditionals-with-match-guards) works:

```rust
    .when(|e| {
        matches!(e, janus_collector::Error::Http(error_response) if error_response.dap_problem_type() == Some(&DapProblemType::InvalidBatchSize))
    })
```

It's not much prettier after the formatter gets done with it, but it does indent less.

Reply:
I .. was not coming up with that syntax. Clever. Maybe I'll get it right next next time!

Applied in a7c4ff1968a31c3e7c09647eadf1dd6e3a30e3a9